pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -245,11 +245,14 @@ func (c *Controller) syncOwnedProvisionRequest(ctx context.Context, wl *kueue.Wo
 		attempt := int32(1)
 		shouldCreatePr := false
 		if exists {
-			if isFailed(oldPr) {
+			if isFailed(oldPr) || isBookingExpired(oldPr) {
+				if isBookingExpired(oldPr) && workload.IsAdmitted(wl) {
```

Comment:
```suggestion
			if isFailed(oldPr) || (isBookingExpired(oldPr) && workload.IsAdmitted(wl)){
```

I think that this break point is not needed since we can express the same conditional by this, right?

Reply:
In general, reducing the nested conditional would improve cognitive load.